autopilot-daemon/pkg/cmd/main.go

```diff
@@ -10,12 +11,45 @@ import (
 
 	"github.com/IBM/autopilot/pkg/handlers"
 	"github.com/IBM/autopilot/pkg/utils"
-	"k8s.io/klog/v2"
-
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	toolswatch "k8s.io/client-go/tools/watch"
+	"k8s.io/klog/v2"
 )
 
+func watchNodes() {
+
+	watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
+		timeout := int64(60)
+		fieldselector, err := fields.ParseSelector("metadata.name=" + os.Getenv("NODE_NAME"))
+		if err != nil {
+			klog.Info("Error in creating the field selector", err.Error())
+			return nil, err
+		}
+		return utils.GetClientsetInstance().Cset.CoreV1().Nodes().Watch(context.Background(), metav1.ListOptions{TimeoutSeconds: &timeout, FieldSelector: fieldselector.String()})
+	}
+
+	watcher, _ := toolswatch.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watchFunc})
+
+	for event := range watcher.ResultChan() {
```

Comment:
What sort of actions would trigger this event beyond an updated label?

Reply:
Watchers are on all CRUD events on objects, we just implement the modify in this case.
So this will be triggered where the object changes in some way, and we just re-read the labels

Reply:
`watchNode` feels right to me